cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_nodegroup_test.go

```diff
@@ -125,12 +126,34 @@ func TestNodeGroupNewNodeGroup(t *testing.T) {
 			return
 		}
 
-		if ng.Name() != tc.name {
-			t.Errorf("expected %q, got %q", tc.name, ng.Name())
+		if ng == nil {
+			t.Fatal("expected nodegroup to be non-nil")
+		}
+
+		var expectedName string
+		var expectedNamespace string
+
+		switch v := (ng.scalableResource).(type) {
+		case *machineSetScalableResource:
+			expectedName = testConfig.spec.machineSetName
+			expectedNamespace = testConfig.spec.namespace
```

Comment:
If the namespace is independent of the scalable resource, it can go out of the `switch` block.